sdl2_ttf 2.24.0 cause VCMI 1.6.2 compilation error

**Describe the bug**
After upgrading sdl2_ttf from version 2.22.0 to 2.24.0, the VCMI 1.6.2 package does not compile correctly with Clang. I see this error.

After downgrading sdl2_ttf to 2.22.0, the compilation works fine.

```
In file included from /builddir/build/BUILD/vcmi-1.6.2-build/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:23:
DEBUG util.py:461:  /usr/include/SDL2/SDL_ttf.h:165:16: [0;1;31merror: type alias 'TTF_Font' cannot be referenced with the 'struct' specifier
DEBUG util.py:461:    165 | typedef struct TTF_Font TTF_Font;
DEBUG util.py:461:        | [0;1;32m               ^
DEBUG util.py:461:  /builddir/build/BUILD/vcmi-1.6.2-build/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.h:20:7: [0;1;36mnote: declared here
DEBUG util.py:461:     20 | using TTF_Font = struct _TTF_Font;
DEBUG util.py:461:        | [0;1;32m      ^
DEBUG util.py:461:  1 error generated.
```

Full build log: http://file-store.openmandriva.org/api/v1/file_stores/7f69c23336b609c0c4e308461ce899b62efa34d5.log?show=true

**To Reproduce**
Steps to reproduce the behavior:

1. Install sdl2_ttf 2.24.0 devel
2. Try compile VCMI 1.6.2 with it
3. See error

- OS: [e.g. Windows, macOS Intel, macOS ARM, Android, Linux, iOS] Linux
- Version: [VCMI version] 1.6.2

> ```diff
> -using TTF_Font = struct _TTF_Font;
> +typedef struct _TTF_Font TTF_Font;
> ```

```
In file included from /data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:23:
/usr/include/SDL2/SDL_ttf.h:165:16: error: using typedef-name ‘TTF_Font’ after ‘struct’
  165 | typedef struct TTF_Font TTF_Font;
      |                ^~~~~~~~
[ 89%] Building CXX object client/CMakeFiles/vcmiclientcommon.dir/globalLobby/GlobalLobbyLoginWindow.cpp.o
In file included from /data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:11:
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.h:20:26: note: ‘TTF_Font’ has a previous declaration here
   20 | typedef struct _TTF_Font TTF_Font;
      |                          ^~~~~~~~
/usr/include/SDL2/SDL_ttf.h:165:25: error: conflicting declaration ‘typedef int TTF_Font’
  165 | typedef struct TTF_Font TTF_Font;
      |                         ^~~~~~~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.h:20:26: note: previous declaration as ‘typedef struct _TTF_Font TTF_Font’
   20 | typedef struct _TTF_Font TTF_Font;
```

> please read the above comments

Yes, I've read it. It is not working without `using TTF_Font = struct _TTF_Font;` too.

```
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:42:12: error: no declaration matches ‘TTF_Font* CTrueTypeFont::loadFont(const JsonNode&)’
   42 | TTF_Font * CTrueTypeFont::loadFont(const JsonNode &config)
      |            ^~~~~~~~~~~~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:42:12: note: no functions named ‘TTF_Font* CTrueTypeFont::loadFont(const JsonNode&)’
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.h:21:7: note: ‘class CTrueTypeFont’ defined here
   21 | class CTrueTypeFont final : public IFont
      |       ^~~~~~~~~~~~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp: In constructor ‘CTrueTypeFont::CTrueTypeFont(const JsonNode&)’:
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:66:14: error: ‘loadFont’ was not declared in this scope
   66 |         font(loadFont(fontConfig), TTF_CloseFont),
      |              ^~~~~~~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:66:9: error: expression list treated as compound expression in mem-initializer [-fpermissive]
   66 |         font(loadFont(fontConfig), TTF_CloseFont),
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:73:31: error: request for member ‘get’ in ‘((CTrueTypeFont*)this)->CTrueTypeFont::font’, which is of non-class type ‘const int’
   73 |         TTF_SetFontStyle(font.get(), getFontStyle(fontConfig));
      |                               ^~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:74:33: error: request for member ‘get’ in ‘((CTrueTypeFont*)this)->CTrueTypeFont::font’, which is of non-class type ‘const int’
   74 |         TTF_SetFontHinting(font.get(),TTF_HINTING_MONO);
      |                                 ^~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:79:62: error: request for member ‘get’ in ‘((CTrueTypeFont*)this)->CTrueTypeFont::font’, which is of non-class type ‘const int’
   79 |                                          TTF_FontHeight(font.get()),
      |                                                              ^~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:80:62: error: request for member ‘get’ in ‘((CTrueTypeFont*)this)->CTrueTypeFont::font’, which is of non-class type ‘const int’
   80 |                                          TTF_FontAscent(font.get()),
      |                                                              ^~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:81:63: error: request for member ‘get’ in ‘((CTrueTypeFont*)this)->CTrueTypeFont::font’, which is of non-class type ‘const int’
   81 |                                          TTF_FontDescent(font.get()),
      |                                                               ^~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:82:64: error: request for member ‘get’ in ‘((CTrueTypeFont*)this)->CTrueTypeFont::font’, which is of non-class type ‘const int’
   82 |                                          TTF_FontLineSkip(font.get())
      |                                                                ^~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp: In member function ‘virtual size_t CTrueTypeFont::getFontAscentScaled() const’:
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:90:36: error: request for member ‘get’ in ‘((const CTrueTypeFont*)this)->CTrueTypeFont::font’, which is of non-class type ‘const int’
   90 |         return TTF_FontAscent(font.get());
      |                                    ^~~
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp: In member function ‘virtual size_t CTrueTypeFont::getLineHeightScaled() const’:
/data/linux/bb/vcmi/src/vcmi-1.6.2/client/renderSDL/CTrueTypeFont.cpp:95:36: error: request for member ‘get’ in ‘((const CTrueTypeFont*)this)->CTrueTypeFont::font’, which is of non-class type ‘const int’
   95 |         return TTF_FontHeight(font.get());
   ```

--- a/client/renderSDL/CTrueTypeFont.cpp
+++ b/client/renderSDL/CTrueTypeFont.cpp
@@ -20,8 +20,6 @@
 #include "../../lib/filesystem/Filesystem.h"
 #include "../../lib/texts/TextOperations.h"
 
-#include <SDL_ttf.h>
-
 std::pair<std::unique_ptr<ui8[]>, ui64> CTrueTypeFont::loadData(const JsonNode & config)
 {
 	std::string filename = "Data/" + config["file"].String();

--- a/client/renderSDL/CTrueTypeFont.h
+++ b/client/renderSDL/CTrueTypeFont.h
@@ -11,13 +11,13 @@
 
 #include "../render/IFont.h"
 
+#include <SDL_ttf.h>
+
 VCMI_LIB_NAMESPACE_BEGIN
 class JsonNode;
 VCMI_LIB_NAMESPACE_END
 
 class CBitmapFont;
-
-using TTF_Font = struct _TTF_Font;
 
 class CTrueTypeFont final : public IFont
 {

